src/history.js

```diff
@@ -187,14 +187,14 @@ class History extends Emitter {
     if (this.size <= 0) {
       this.begin()
       return
-    } else if (!next) {
+    } else if (action === this.head) {
       next = this.head = parent
     } else if (action === this.root) {
       this.root = next
     }
 
     if (!action.disabled) {
-      this.reconcile(next)
+      this.reconcile(this.head)
```

Comment:
Ahhhh. I see now. Okay. It sounds like we need to know if the action is within the current branch. We could do that two ways:

```
// get the active branch
let isActive = this.toArray().indexOf(next) >= 0

// Traverse to the head
let isActive = false
let cursor = next
while (cursor) {
  if (cursor === head) {
    isActive = true
    break;
  }
  cursor = cursor.next
}
```

The later is probably more performant, but I'm not sure how much it really matters.